ui/src/main/java/edu/wpi/grip/ui/analysis/AnalysisWindowController.java

```diff
@@ -0,0 +1,339 @@
+      numRecentSamples = numRuns;
+      benchmarker.run(numRuns);
+    }
+  }
+
+  @FXML
+  private void exportCsv() {
+    // Show benchmarking results
+    Platform.runLater(() -> {
+      Alert a = new Alert(Alert.AlertType.INFORMATION);
+      a.setHeaderText("Benchmarking results");
+      TextArea resultArea = new TextArea(csvReport);
+      a.getDialogPane().setContent(resultArea);
+      a.showAndWait();
+    });
+  }
+
+  /**
+   * Streams the entries in a {@code Map<Step, *>} sorted by the step's index according to the
+   * {@link #stepIndexer}.
+   *
+   * @param m   the map to stream
+   * @param <E> the type of the values in the map
+   * @return a stream of the entries in the map, sorted by their key's index.
+   */
+  private <E> Stream<Map.Entry<Step, E>> sortedStream(Map<Step, E> m) {
+    return m.entrySet()
+        .stream()
+        .sorted((e1, e2) -> stepIndexer.compare(e1.getKey(), e2.getKey()));
+  }
+
+  /**
+   * Creates a CSV report of the most recent benchmark.
+   */
+  private String createReport() {
+    StringBuilder sb = new StringBuilder(CSV_REPORT_HEADER);
+
+    // List of step names, in the order they run in
+    final List<String> stepNames = sortedStream(sampleMap)
+        .map(Map.Entry::getKey)
+        .map(Step::getOperationDescription)
+        .map(OperationDescription::name)
+        .collect(Collectors.toList());
+
+    // Statistics of each step's run time
+    final List<Statistics> statistics = sortedStream(sampleMap)
+        .map(Map.Entry::getValue)
+        .map(s -> s.stream().mapToDouble(Long::doubleValue).toArray())
+        .map(Statistics::of)
+        .collect(Collectors.toList());
+
+    // Average run times for each step
+    final List<Double> averageRunTimes = statistics
+        .stream()
+        .map(Statistics::getMean)
+        .map(t -> t / 1000) // convert us to ms
+        .collect(Collectors.toList());
+
+    // Average total run time for the whole pipeline
+    final double averageTotalRunTime = statistics
+        .stream()
+        .mapToDouble(Statistics::getMean)
+        .map(t -> t / 1000) // convert us to ms
+        .sum();
+
+    // What percent of the total each step took
+    final List<Double> percentRunTimes = averageRunTimes
+        .stream()
+        .map(t -> 100 * t / averageTotalRunTime)
+        .collect(Collectors.toList());
+
+    // Standard deviation in run times for each step
+    final List<Double> stddev = statistics
+        .stream()
+        .map(Statistics::getStandardDeviation)
+        .map(t -> t / 1000) // convert us to ms
+        .collect(Collectors.toList());
+    for (int i = 0; i < statistics.size(); i++) {
+      sb.append(stepNames.get(i));
+      sb.append(',');
+      sb.append(percentRunTimes.get(i));
+      sb.append(',');
+      sb.append(averageRunTimes.get(i));
+      sb.append(',');
+      sb.append(stddev.get(i));
+      sb.append('\n');
+    }
+    return sb.toString();
```

Comment:
I have a plugin from codecov that shows me in the PR what code is covered and what isn't.
This isn't covered by tests.
https://chrome.google.com/webstore/detail/codecov-extension/keefkhehidemnokodkdkejapdgfjmijf/related